### ISSUE TITLE
Non-residual datatypes

### ISSUE THREAD

**Issue by vallis** [nanograv/enterprise#106]
We should be thinking about how we will handle datatypes different than residuals, e.g., DM data for cyclic spectroscopy timing (with DM uncertainties possibly correlated with TOA uncertainties), profile intensity data for profile-domain timing, etc.

`Pulsar` can easily host other vectors, but we will probably have to extend/replace the `PTA` object.

**Comment by jellis18** [comment]
Yes this is definitely true and I've been a bit worried about this as we have been going along. I know what we need for the DM data but we would have to work with Lindley for the profile domain stuff because I have no idea what is needed for that.
